### PR TITLE
Fix dead link in build docker environment in developer guide

### LINKED ISSUE #8378: [Enhancement] [doc] there are some dead link in fe doc pages
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.

### Description

when i read doris developer docs, i found some dead link:
1. link for [problems with mounting] in https://doris.incubator.apache.org/developer-guide/docker-dev.html
2. link for [挂载的问题] in https://doris.incubator.apache.org/zh-CN/developer-guide/docker-dev.html

i think change the host in link from  doris.apache.org to doris.incubator.apache.org should fix it

### Solution

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by morrySnow**
# Proposed changes

Issue Number: close #8378

## Problem Summary:

Describe the overview of changes.

## Checklist(Required)

1. Does it affect the original behavior: (No)
2. Has unit tests been added: (No Need)
3. Has document been added or modified: (Yes)
4. Does it need to update dependencies: (No)
5. Are there any changes that cannot be rolled back: (No)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Review by dataroaring — DISMISSED** [review]

**Review by hf200012 — DISMISSED** [review]
LGTM

File: docs/en/developer-guide/docker-dev.md [hunk 1]
```diff
@@ -90,7 +90,7 @@ docker build -t doris .
 
 run image
 
-note! [problems with mounting](https://doris.apache.org/master/zh-CN/installing/compilation.html#%E4%BD%BF%E7%94%A8-docker-%E5%BC%80%E5%8F%91%E9%95%9C%E5%83%8F%E7%BC%96%E8%AF%91-%E6%8E%A8%E8%8D%90)
+note! [problems with mounting](/installing/compilation.html#developing-mirror-compilation-using-docker-recommended)
```
Comment (morningman) [inline docs/en/developer-guide/docker-dev.md:93]:
```suggestion
note! [problems with mounting](../installing/compilation.md)
```

File: docs/zh-CN/developer-guide/docker-dev.md [hunk 2]
```diff
@@ -90,7 +90,7 @@ docker build -t doris .
 
 运行镜像
 
-此处按需注意 [挂载的问题](https://doris.apache.org/master/zh-CN/installing/compilation.html#%E4%BD%BF%E7%94%A8-docker-%E5%BC%80%E5%8F%91%E9%95%9C%E5%83%8F%E7%BC%96%E8%AF%91-%E6%8E%A8%E8%8D%90)
+此处按需注意 [挂载的问题](/zh-CN/installing/compilation.html#%E4%BD%BF%E7%94%A8-docker-%E5%BC%80%E5%8F%91%E9%95%9C%E5%83%8F%E7%BC%96%E8%AF%91-%E6%8E%A8%E8%8D%90)
```
Comment (morningman) [inline docs/zh-CN/developer-guide/docker-dev.md:93]:
```suggestion
此处按需注意 [挂载的问题](../installing/compilation.md)
```

File: docs/en/developer-guide/docker-dev.md (hunk 1 above)
Comment (morrySnow) [reply docs/en/developer-guide/docker-dev.md:93]: ok, i'll submit a new commit right now

**Review by EmmyMiao87 — APPROVED** [review]
LGTM

**Pull request merged**